### PR TITLE
Move cumulus zombienet tests to aura & async backing

### PR DESCRIPTION
**Pull request by skunert**
Cumulus test-parachain node and test runtime were still using relay chain consensus and 12s blocktimes. With async backing around the corner on the major chains we should switch our tests too.

Also needed to nicely test the changes coming to collators in #3168.

### Changes Overview
- Followed the [migration guide](https://wiki.polkadot.network/docs/maintain-guides-async-backing) for async backing for the cumulus-test-runtime
- Adjusted the cumulus-test-service to use the correct import-queue, lookahead collator etc.
- The block validation function now uses the Aura Ext Executor so that the seal of the block is validated
- Previous point requires that we seal block before calling into `validate_block`, I introduced a helper function for that
- Test client adjusted to provide a slot to the relay chain proof and the aura pre-digest

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5431707

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5434289

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5434455

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5573768

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5580368

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5582639

File: cumulus/client/consensus/common/src/tests.rs [hunk 1]
```diff
@@ -777,10 +778,11 @@ fn prune_blocks_on_level_overflow() {
 
 #[test]
 fn restore_limit_monitor() {
+	sp_tracing::try_init_simple();
```
Comment (lexnv) [inline cumulus/client/consensus/common/src/tests.rs:781]: dq: Was this used for debugging?

**Review by lexnv — APPROVED** [review]
LGTM based on my limited knowledge of cumulus testing 👍

File: cumulus/test/client/src/block_builder.rs [hunk 2]
```diff
@@ -32,12 +37,19 @@ pub trait InitBlockBuilder {
...
 	fn init_block_builder(
 		&self,
 		validation_data: Option<PersistedValidationData<PHash, PBlockNumber>>,
 		relay_sproof_builder: RelayStateSproofBuilder,
-	) -> sc_block_builder::BlockBuilder<Block, Client>;
+	) -> (
+		sc_block_builder::BlockBuilder<Block, Client>,
+		PersistedValidationData<PHash, PBlockNumber>,
+		Slot,
+	);
```
Comment (michalkucharczyk) [inline cumulus/test/client/src/block_builder.rs:52]: struct would be nice here. the tuple seems to be duplicated.

File: cumulus/test/service/src/lib.rs [hunk 3]
```diff
@@ -427,59 +424,61 @@ where
...
-			announce_block,
-			runtime_api: client.clone(),
-			spawner: task_manager.spawn_handle(),
-			para_id,
-			parachain_consensus,
-			key: collator_key,
-			overseer_handle,
-		})
-		.await;
+		if let Consensus::Null = consensus {
```
Comment (michalkucharczyk) [inline cumulus/test/service/src/lib.rs:427]: Not really related to this PR: why do we need Null consensus? When (and why) is it used?

File: cumulus/pallets/parachain-system/src/validate_block/tests.rs [hunk 4]
```diff
@@ -316,15 +322,15 @@ fn validate_block_works_with_child_tries() {
...
 		&client,
 		vec![generate_extrinsic(&client, Alice, TestPalletCall::read_and_write_child_tries {})],
 		parent_head.clone(),
 		Default::default(),
 	);
 
-	let header = block.header().clone();
-
+	let block = seal_block(block, slot, &client);
+	let header: cumulus_test_runtime::Header = block.header().clone();
```
Comment (michalkucharczyk) [inline cumulus/pallets/parachain-system/src/validate_block/tests.rs:333]: Can type be removed? Was rustc complaining?

File: cumulus/test/runtime/src/lib.rs [hunk 5]
```diff
@@ -74,12 +75,18 @@ pub use test_pallet::Call as TestPalletCall;
...
-	pub struct SessionKeys {}
+	pub struct SessionKeys {
+		pub aura: Aura,
+	}
 }
 
 /// The para-id used in this runtime.
 pub const PARACHAIN_ID: u32 = 100;
 
+const UNINCLUDED_SEGMENT_CAPACITY: u32 = 3;
```
Comment (michalkucharczyk) [inline cumulus/test/runtime/src/lib.rs:86]: Curiosity question: The async-backing-upgrade guide says 2-3. Why 3?

**Review by michalkucharczyk — APPROVED** [review]

File: cumulus/test/service/src/lib.rs [hunk 6]
```diff
@@ -360,12 +358,13 @@ where
 			spawn_handle: task_manager.spawn_handle(),
 			relay_chain_interface: relay_chain_interface.clone(),
 			import_queue: params.import_queue,
-			sybil_resistance_level: CollatorSybilResistance::Unresistant, // no consensus
+			sybil_resistance_level: CollatorSybilResistance::Resistant, // no consensus
```
Comment (bkchr) [inline cumulus/test/service/src/lib.rs:361]:
```suggestion
			sybil_resistance_level: CollatorSybilResistance::Resistant, // either Aura that is resistant or null that is not producing any blocks at all.
```

File: cumulus/test/runtime/src/lib.rs [hunk 7]
```diff
@@ -271,8 +320,22 @@ impl cumulus_pallet_parachain_system::Config for Runtime {
...
+	type DisabledValidators = ();
+	type MaxAuthorities = ConstU32<32>;
+	type AllowMultipleBlocksPerSlot = ConstBool<true>;
+	type SlotDuration = ConstU64<SLOT_DURATION>;
+}
+
+impl pallet_authorship::Config for Runtime {
+	type FindAuthor = pallet_session::FindAccountFromAuthorIndex<Self, Aura>;
+	type EventHandler = (CollatorSelection,);
 }
```
Comment (bkchr) [inline cumulus/test/runtime/src/lib.rs:339]: We don't really need this pallet.

File: cumulus/test/runtime/src/lib.rs [hunk 8]
```diff
@@ -293,6 +356,11 @@ construct_runtime! {
 		TransactionPayment: pallet_transaction_payment,
 		TestPallet: test_pallet,
 		Glutton: pallet_glutton,
+		Authorship: pallet_authorship,
+		CollatorSelection: pallet_collator_selection,
+		Session: pallet_session,
```
Comment (bkchr) [inline cumulus/test/runtime/src/lib.rs:361]: Same for these two. We have hard coded authorities any way? Aka they are not changing.

File: cumulus/test/runtime/Cargo.toml [hunk 9]
```diff
@@ -77,3 +93,5 @@ std = [
 	"substrate-wasm-builder",
 ]
 increment-spec-version = []
+
+experimental = ["pallet-aura/experimental"]
```
Comment (bkchr) [inline cumulus/test/runtime/Cargo.toml:97]:
```suggestion
```

This feature doesn't do anything anymore. Please also remove it from the pallet Cargo.toml

File: cumulus/pallets/parachain-system/src/validate_block/tests.rs (hunk 4 above)
Comment (skunert) [reply cumulus/pallets/parachain-system/src/validate_block/tests.rs]: Yep, removed!

File: cumulus/test/client/src/block_builder.rs (hunk 2 above)
Comment (skunert) [reply cumulus/test/client/src/block_builder.rs]: Done!

File: cumulus/test/service/src/lib.rs (hunk 3 above)
Comment (skunert) [reply cumulus/test/service/src/lib.rs:430]: We use it in the [pov-recovery zombienet](https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/zombienet/tests/0002-pov_recovery.toml#L42) test to have a authority that is not producing blocks.  Triggers different codepaths with reduced recovery delay.

**Comment by skunert** [comment]
Hmm this breaks our warpsync zombienet test, I will generate new snapshots.

**Pull request merged**